magda-storage-api/src/createApiRouter.ts

```diff
@@ -29,6 +29,66 @@ export default function createApiRouter(options: ApiRouterOptions) {
         return res.status(200).send("OK");
     });
 
+    /**
+     * @apiGroup Storage
+     *
+     * @api {POST} /v0/create/bucket?bucket={bucket} Request to create a new bucket
+     *
+     * @apiDescription Creates a new bucket with a specified name. Restricted to admins only.
+     *
+     * @apiParam (Request query) {string} bucket The name of the bucket to be created
+     *
+     * @apiSuccessExample {json} 200
+     *    {
+     *        "message":"Bucket my-bucket created successfully in unspecified-region 🎉"
+     *    }
+     *
+     * @apiErrorExample {json} 400
+     *    {
+     *        "message": "Bucket my-bucket already exists 👍"
```

Comment:
I'm not sure this should be a `400`, given that it's not really a bad request per se - I think it's pretty reasonable to do this multiple times (e.g. on startup) just to make sure a bucket is there right? If this returns a 400 then when this is called, clients will have to have special error handling logic in order to cope with the bucket already existing, which is usually OK.

Reply:
There should probably be a `400` response if you don't provide an id at all though!

Reply:
No ID at all will now result in hitting an invalid endpoint. Trying to create a bucket that exists will return 201 still but with a different message.